src/foam/nanos/rope/ROPEUtils.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.rope',
+  name: 'NextRelationshipsList',
+
+  properties: [
+    {
+      name: 'nextRelationships',
+      class: 'List',
+      javaType: 'java.util.List<String>'
+    }
+  ]
+
+});
+
+foam.CLASS({
+  package: 'foam.nanos.rope',
+  name: 'RelationshipMap',
+
+  properties: [
+    {
+      name: 'map',
+      class: 'Map',
+      javaType: 'java.util.Map<String, NextRelationshipsList>'
+    }
+  ],
+
+  methods: [
+    {
+      name: 'get',
+      javaType: 'java.util.List<String>',
+      args: [
+        { name: 'relationshipKey', class: 'String' }
+      ],
+      javaCode: `
+        return getMap().containsKey(relationshipKey) && getMap().get(relationshipKey) != null ? getMap().get(relationshipKey).getNextRelationships() : null;
+      `
+    },
+  ]
+
+});
+
+foam.CLASS({
+  package: 'foam.nanos.rope',
+  name: 'CRUDMap',
+
+  javaImports: [
+    'java.util.Map',
+    'java.util.List'
+  ],
+
+  properties: [
+    {
+      name: 'create',
+      class: 'Map',
+      javaType: 'java.util.Map<String, NextRelationshipsList>'
+    },
+    {
+      name: 'read',
+      class: 'Map',
+      javaType: 'java.util.Map<String, NextRelationshipsList>'
+    },
+    {
+      name: 'update',
+      class: 'Map',
+      javaType: 'java.util.Map<String, NextRelationshipsList>',
+    },
+    {
+      name: 'delete',
+      class: 'Map',
+      javaType: 'java.util.Map<String, NextRelationshipsList>',
+    }
+  ],
+  
+  axioms: [
+    {
+      name: 'javaExtras',
+      buildJavaClass: function (cls) {
+        cls.extras.push(`
+          /**
+           * returns the map for a crud operation
+           * */
+          public Map<String, NextRelationshipsList> get(String crudKey) {
+            switch ( crudKey ) {
+              case "create":
```

Comment:
Should probably make these into an enum and use that instead of a string for `crudKey`. Then user's can't pass in an invalid value.

Reply:
modelled crudoperations as enums in this commit:
https://github.com/noodlemoodle/foam2/pull/41/commits/60db828106850fba14633cfb9d7d4fef6dcd8cb0